Failed ScalaUdpMulticastSpec

There was a failure on [akka-nightly](https://jenkins.akka.io:8498/job/akka-nightly/1701/)
```
[info] ScalaUdpMulticastSpec:
[info] listener
[info] - should send message back to sink *** FAILED ***
[info]   java.lang.AssertionError: assertion failed: timeout (3 seconds) during expectMsgClass waiting for class akka.io.Udp$Bound
[info]   at scala.Predef$.assert(Predef.scala:179)
[info]   at akka.testkit.TestKitBase$class.expectMsgClass_internal(TestKit.scala:433)
[info]   at akka.testkit.TestKitBase$class.expectMsgType(TestKit.scala:406)
[info]   at akka.testkit.TestKit.expectMsgType(TestKit.scala:733)
[info]   at docs.io.ScalaUdpMulticastSpec$$anonfun$1$$anonfun$apply$mcV$sp$1.apply$mcV$sp(ScalaUdpMulticastSpec.scala:36)
[info]   at docs.io.ScalaUdpMulticastSpec$$anonfun$1$$anonfun$apply$mcV$sp$1.apply(ScalaUdpMulticastSpec.scala:21)
[info]   at docs.io.ScalaUdpMulticastSpec$$anonfun$1$$anonfun$apply$mcV$sp$1.apply(ScalaUdpMulticastSpec.scala:21)
[info]   at org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
[info]   at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   ...
```

Previous issues with this test (#15772, #15785) seem to be unrelated.

--- a/akka-docs/rst/java/code/docs/io/JavaUdpMulticastTest.java
+++ b/akka-docs/rst/java/code/docs/io/JavaUdpMulticastTest.java
@@ -35,7 +35,9 @@ public class JavaUdpMulticastTest {
             for (Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces(); interfaces.hasMoreElements() && ipv6Iface == null;) {
                 NetworkInterface interf = interfaces.nextElement();
                 // awdl0 is a special interface on OSX that we cannot use
-                if (!interf.getDisplayName().equals("awdl0")) {
+                if (!interf.getDisplayName().equals("awdl0") &&
+                        // we do not want to use virtual docker interfaces
+                        !interf.getDisplayName().contains("docker")) {
                     for (Enumeration<InetAddress> addresses = interf.getInetAddresses(); addresses.hasMoreElements() && ipv6Iface == null; ) {
                         InetAddress address = addresses.nextElement();
                         if (address instanceof Inet6Address) {

--- a/akka-docs/rst/scala/code/docs/io/ScalaUdpMulticastSpec.scala
+++ b/akka-docs/rst/scala/code/docs/io/ScalaUdpMulticastSpec.scala
@@ -19,10 +19,15 @@ class ScalaUdpMulticastSpec extends TestKit(ActorSystem("ScalaUdpMulticastSpec")
 
   "listener" should {
     "send message back to sink" in {
-      val Some(ipv6Iface) = NetworkInterface.getNetworkInterfaces.collectFirst {
-        // awdl0 is a special interface on OSX that we cannot use
-        case iface if iface.getInetAddresses.exists(_.isInstanceOf[Inet6Address]) && iface.getDisplayName != "awdl0" => iface
+
+      def okInterfaceToUse(iface: NetworkInterface): Boolean = {
+        iface.getInetAddresses.exists(_.isInstanceOf[Inet6Address]) &&
+          // awdl0 is a special interface on OSX that we cannot use
+          iface.getDisplayName != "awdl0" &&
+          // we do not want to use virtual docker interfaces
+          !iface.getDisplayName.contains("docker")
       }
+      val Some(ipv6Iface) = NetworkInterface.getNetworkInterfaces.find(okInterfaceToUse)
 
       // host assigned link local multicast address http://tools.ietf.org/html/rfc3307#section-4.3.2
       // generate a random 32 bit multicast address with the high order bit set

